app/models/game.rb

```diff
@@ -30,4 +30,37 @@ def populate_game!
   def piece_at(x,y)
     pieces.find_by(x_coord: x, y_coord: y) || EmptySquare.new
   end
+
+  def in_check? 
+    if white_in_check == true || black_in_check == true
+      return true 
+    else 
+      return false
+    end
+  end
+
+  def white_in_check
+    self.pieces.all.each do |piece|
```

Comment:
@Brandongoodman615 remember the `any?` method we used in the obstructions ( if I remember correctly) https://coderwall.com/p/a1pspg/ruby-methods-all-vs-any. It would be a great candidate to apply here.. We want to check if 'any' of the other pieces would endanger the king... Same goes for the 'black' method...